spring-integration-redis/src/main/java/org/springframework/integration/redis/inbound/RedisQueueMessageDrivenEndpoint.java

```diff
@@ -158,6 +160,16 @@ public void setRecoveryInterval(long recoveryInterval) {
 		this.recoveryInterval = recoveryInterval;
 	}
 
+	/**
+	 * Should data from the Redis queue be read from the left end of the queue (using leftPop)?
+	 * This is useful when the queue is fed by a software that uses rightPush.
```

Comment:
Yes, we must not be so specific in the JavaDocs. This option is so general like we just should say here how it changes the behavior of the components. And that's all.

We may just mention such a possible scenario in the Reference Manual (it should be addressed with this PR as well - `redis.adoc`).